ftprime/argrecorder.py

```diff
@@ -167,20 +170,29 @@ def __str__(self):
         # ret += "\n---------\n"
         return ret
 
-    def __call__(self, parent, time, population, child, left, right):
+    def __call__(self, parents, times, populations, childrens, lefts, rights):
         """
-        Does both ``add_individual()`` and ``add_record steps()``.
-
-        :param int parent: Input ID of the parent.
-        :param float time: The time of birth of the child.
-        :param int population: The population ID where the child is born.
-        :param int child: Input ID of the child.
-        :param float left: Left end of the segment that child inherits from parent.
-        :param float right: Right end of the segment that child inherits from parent.
+        Does both ``add_individuals()`` and ``add_records()`` steps.
+
+        :param iterable of int parent: The input IDs of the parent.
+        :param iterable of int populations: The population ID of birth of the
+            children.
+        :param iterable of float times: The time of birth of the children.
+        :param iterable of tuple childrens: An iterable of input IDs of the
```

Comment:
this would be simpler if we changed this to be `edge`-like rather than `edgeset`-like, passing `childs` instead of `childrens`.